packages/material-ui/src/Button/Button.js

```diff
@@ -226,6 +227,18 @@ function Button(props) {
     ...other
   } = props;
 
+  warning(
+    process.env.MUI_SUPPRESS_DEPRECATION_WARNINGS || variant !== 'flat',
```

Comment:
Idea: pull out our old deprecated prop types [from the grave](https://github.com/mui-org/material-ui/blob/v0.x/src/utils/deprecatedPropType.js) 💀 and use it. As you see fit, maybe it can greatly help locate the origin of the warning.

Reply:
That looks nice, thanks. Though it looks like this marks the hole prop as deprecated. I would need to do something like `oneOfType(oneOf(goodVariants), deprecated(oneOf(deprecated)))`? This might break the docs?

Reply:
It used to work with all our stack.

Reply:
@oliviertassinari - the problem is that we aren't deprecating a prop, only some of the accepted values.

Reply:
Let's start simple. The amount of work vs the outcome probably doesn't worse it. We will see.